### PR TITLE
CMP-3818: Add test to test tokenrules pass when configuring oauthclients

### PR DESCRIPTION
**Pull request by Anna-Koudelkova**
In the upstream content testing we apply remediation for rules `ocp4-oauth-or-oauthclient-token-maxage` and `ocp4-oauth-or-oauthclient-inactivity-timeout`  by patching it on cluster level. Both rules should be passing if client do not set the value on a cluster level, but set the values for all oauthclients individually. 
This PR should add test to see if both rules are passing when the values are set per each client. 

```
2026/01/26 13:04:23 All scans in ComplianceSuite have finished (test-token-rules-pass-oauth-clients-configurable)
--- PASS: TestTokenRulesPassOauthClientsConfigurable (86.17s)
PASS
...
ok  	github.com/ComplianceAsCode/compliance-operator/tests/e2e/serial	298.255s
```

Note: I have run into some issues while trying to make this testcase work - mainly because I understood the description of the rule `ocp4-oauth-or-oauthclient-token-maxage` that for it to pass it is enough to set the values for all oauth clients - and I can set whatever value I want. Which is not true - the values set need to correspond with the value in the corresponding variable `ocp4-var-oauth-token-maxage`. If the values from the variable and from the oauth clients mismatch, the rule would FAIL. Is this the expected behavior?

### CODE REVIEW

**Review by rhmdnd — CHANGES_REQUESTED** [review]
A couple suggestions on reducing calls and a potential approach to cleaning up after the test is done, since we're dealing with a cluster-wide setting.

File: tests/e2e/framework/common.go [hunk 1]
```diff
@@ -3103,3 +3108,137 @@ func (f *Framework) waitForNamespaceDeletion(namespace string, retryInterval, ti
...
+		}
+		metadata, ok := clientObj["metadata"].(map[string]interface{})
+		if !ok {
+			continue
+		}
+		clientName, ok := metadata["name"].(string)
+		if !ok {
+			continue
+		}
+		maxAge := int64(0)
```
Comment (rhmdnd) [inline tests/e2e/framework/common.go]: If an OAuth client doesn't have `maxAge` or `inactivity` set, the config passed back to the restore function will set them to 0 instead of leaving them unset.
For example:                                                                                                                                                                                             

```yaml
# before test
OAuth Config:
  name: foo

# during test
OAuth Config:
  name: foo
  maxAge: 28800
  inactivity: 600

# after restore
OAuth Config:
  name: foo
  maxAge: 0
  inactivity: 0
```

So the restore would set the values to 0 rather than removing them. The question is whether the OAuth server treats an unset value the same as 0.

Either way, this is probably fine given we don't test OAuth outside of this case, but wanted to mention it as a potential edge case in the restoration flow.

File: tests/e2e/framework/common.go [hunk 2]
```diff
@@ -3103,3 +3108,137 @@ func (f *Framework) waitForNamespaceDeletion(namespace string, retryInterval, ti
...
+type OAuthClientConfig struct {
+	Name                                string
+	AccessTokenMaxAgeSeconds            int64
+	AccessTokenInactivityTimeoutSeconds int64
+}
+
+// GetAndPatchAllOAuthClientConfigs lists all OAuth clients once; for each client it saves the current config
+// into the returned slice and then patches the client with the given values. Return the slice to
+// RestoreOAuthClientConfigs when done.
+func (f *Framework) GetAndPatchAllOAuthClientConfigs(maxAgeSeconds, inactivityTimeoutSeconds int64) ([]OAuthClientConfig, error) {
```
Comment (rhmdnd) [inline tests/e2e/framework/common.go]: To your question about combining these. The primary reason behind my previous comment was to advocate for a single patch call for two values instead of two separate patch calls per config (which you've implemented).

I'm not opposed to what you've done here, since it saves off the original configs and patches each client into the same loop.

Alternatively, we could have a `getOAuthClientConfigs` and an `updateOAuthClientConfigs` separately, but in that case we'd be iterating the list of configs three times (getting the original values, updating the clients for the test, restoring the original values) as opposed to just twice (getting the original values and setting the test-specific values, then restoring the original values).

I could go either way - it's up to you how you'd like to structure it and whatever make sense to you organizationally.

**Review by rhmdnd — DISMISSED** [review]
Looks great! Approving pending testing feedback.

/lgtm
/approve

**Comment by vickeybrown** [comment]
It feels a little weird to have all of the OAuth framework stuff in our e2e tests since we're not in the OAuth business, would it make sense to use Openshift's [client-go library](https://github.com/openshift/client-go/tree/master)?

**Comment by Anna-Koudelkova** [comment]
> It feels a little weird to have all of the OAuth framework stuff in our e2e tests since we're not in the OAuth business, would it make sense to use Openshift's [client-go library](https://github.com/openshift/client-go/tree/master)?

I have made a second commit heavily relying on Cursor, and to be honest my experience is not that great to clearly evaluate if this is better or worse - therefore I have kept it as a separate commit. This new version is also passing on a 4.22 cluster, it just feels too bloated to me (but on the other hand there is a structure, so I am really not sure). 

@vickeybrown Could you please have a quick glance? Please shout if this is utter nonsense -  I'd really appreciate it!

**Comment by vickeybrown** [comment]
Looks good to me! Just need to fix merge conflicts

**Review by vickeybrown — APPROVED** [review]
